core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -302,15 +323,37 @@ public void onLoad(Run<?, ?> r) {
     }
 
     private boolean isSafeParameter(String name) {
+        loadSafeParameters();
+        return safeParameters.contains(name);
+    }
+
+    /**
+     * Combines the contents of {@link #SAFE_PARAMETERS_SYSTEM_PROPERTY_NAME}
+     * and {@link #getAdditionalSafeParameters()} into {@link #safeParameters}.
+     * @since TODO
+     */
+    private void loadSafeParameters() {
```

Comment:
🐛  for potential concurrency issues if two threads get into the <code>safeParameters == null</code> at the same time. Should be synchronized